pylossless/pipeline.py

```diff
@@ -798,68 +806,185 @@ def find_breaks(self):
         # to other channels (nonparametric z-score)
         ch_dist = trim_ch_sd - trim_ch_sd.median(dim="ch")
         perc_30 = trim_ch_sd.quantile(0.3, dim="ch")
         perc_70 = trim_ch_sd.quantile(0.7, dim="ch")
-        ch_dist /= perc_70 - perc_30  # shape (chans, time)
+        ch_dist /= perc_70 - perc_30  # shape (chans, epoch)
 
-        mean_ch_dist = ch_dist.mean(dim="time")  # shape (chans)
+        mean_ch_dist = ch_dist.mean(dim="epoch")  # shape (chans)
 
         # find the median and 30 and 70 percentiles
         # of the mean of the channel distributions
         mdn = np.median(mean_ch_dist)
         deviation = np.diff(np.quantile(mean_ch_dist, [0.3, 0.7]))
 
-        bad_ch_names = mean_ch_dist.ch[mean_ch_dist > mdn+6*deviation]
-        self.flagged_chs.add_flag_cat(kind='outliers',
-                                      bad_ch_names=bad_ch_names)
+        return mean_ch_dist.ch[mean_ch_dist > mdn+6*deviation].values.tolist()
 
-        # TODO: Verify: It is unclear this is necessary.
-        # get_epochs() is systematically rereferencing and
-        # all steps (?) uses the get_epochs() function
-        self.flagged_chs.rereference(self.raw)
+    def flag_channels_fixed_threshold(self, threshold=5e-5):
+        """Flag channels based on the stdev value across the time dimension.
+
+        Flags channels if the voltage-variance standard deviation is above
+        the given threshold in n_percent of epochs (default: 20%).
+
+        Parameters
+        ----------
+        threshold : float
+            threshold, in volts. If the standard deviation across time in
+            any channel x epoch indice is above this threshold, then the
+            channel x epoch indices will considered an outlier. Defaults
+            to 5e-5, or 50 microvolts. Note that here, 'time' refers to
+            the samples in an epoch. For each channel, if its std value is
+            above the given threshold in more than 20% of the epochs, it
+            is flagged.
+
+        Notes
+        -----
+        WARNING: the default threshold of 50 microvolts may not be appropriate
+        for a particular dataset or data file, as the baseline voltage variance
+        is affected by the impedance of the system that the data was recorded
+        with. You may need to assess a more appropriate value for your own
+        data.
+        """
+        if 'flag_channels_fixed_threshold' not in self.config:
+            return
+        logger.info('🏃🏃🏃 Starting flag_channels_fixed_threshold !')
+        if 'threshold' in self.config['flag_channels_fixed_threshold']:
+            threshold = (self.config['flag_channels_fixed_threshold']
+                                    ['threshold']
+                         )
+        self._flag_volt_std(flag_dim='ch', threshold=threshold)
+        logger.info('☑️☑️☑️ Done with flag_channels_fixed_threshold!!')
+
+    def flag_epochs_fixed_threshold(self, threshold=5e-5):
```

Comment:
This is a function we coded together (I asked for your help) to flag epochs based on a fixed threshold of the voltage-var stdev (per the lossless paper.)